### PR TITLE
Document the return values of all methods in the docs

### LINKED ISSUE #7337: The docs and the style guide should contain the return value of methods when relevant
* Electron version: All
* Operating system: All

Currently the JSON API doesn't contain return values from methods, this makes converting the JSON to a `.d.ts` file quite difficult as we just assume that methods return `void` or `any`.

I would suggest the return standard being something like this.

```markdown
### `app.getName()`

Returns `String` - The current application's name, which is the name in the application's
`package.json` file.

Usually the `name` field of `package.json` is a short lowercased name, according
to the npm modules spec. You should usually also specify a `productName`
field, which is your application's full capitalized name, and which will be
preferred over `name` by Electron.
```

I.e. If a method has a return value, specify it in the first paragraph of the description in the following format

```markdown
Returns `[TYPE]` - Description of return value
```

/cc @zeke

### PR DESCRIPTION
**Pull request by MarshallOfSound**
As brought up in #7337 methods need to have the return values documented in a standard way in the docs so that `electron-api-docs` can pull out the return values correctly

Fixes #7337

### CODE REVIEW

File: docs/api/browser-window.md [hunk 1]
```diff
@@ -835,7 +849,7 @@ Moves window to `x` and `y`.
 
 #### `win.getPosition()`
 
-Returns an array that contains window's current position.
+Returns `Integer[]` - Contains the window's current position.
```
Comment (zeke) [inline docs/api/browser-window.md]: So this syntax denotes "an array of Integers"? I think I remember seeing something more like `[Integer]` somewhere in the docs, but can't find any examples at the moment. Not saying that is a better syntax, but just want to be sure we're consistent.

File: docs/styleguide.md [hunk 2]
```diff
@@ -101,8 +101,10 @@ Using `autoUpdater` as an example:
 * Constructors must be listed with `###`-level titles.
 * [Static Methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static) must be listed under a `### Static Methods` chapter.
 * [Instance Methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#Prototype_methods) must be listed under an `### Instance Methods` chapter.
+* All methods that have a return value must start their description with "Returns `[TYPE]` - Return description"
 * Instance Events must be listed under an `### Instance Events` chapter.
 * Instance Properties must be listed under an `### Instance Properties` chapter.
+  * Instance properties must start with "A [Property Ttype] ..."
```
Comment (zeke) [inline docs/styleguide.md]: Is `Ttype` a typo?

File: docs/styleguide.md [hunk 3]
```diff
@@ -101,8 +101,10 @@ Using `autoUpdater` as an example:
 * Constructors must be listed with `###`-level titles.
 * [Static Methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static) must be listed under a `### Static Methods` chapter.
 * [Instance Methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#Prototype_methods) must be listed under an `### Instance Methods` chapter.
+* All methods that have a return value must start their description with "Returns `[TYPE]` - Return description"
```
Comment (zeke) [inline docs/styleguide.md]: Might want to add something about how the syntax differs if the return value is an object. (i.e. colon, newline, unordered list)

File: docs/api/browser-window.md (hunk 1 above)
Comment (MarshallOfSound) [inline docs/api/browser-window.md]: I looked around and couldn't see any examples of array values being specified with any format so I went with this format as it is relatively standard as both a declaration in most languages and the typescript definition.

File: docs/styleguide.md (hunk 2 above)
Comment (MarshallOfSound) [inline docs/styleguide.md]: Woops :'(

File: docs/styleguide.md (hunk 3 above)
Comment (MarshallOfSound) [inline docs/styleguide.md]: Yep, will do 👍

File: docs/api/screen.md [hunk 4]
```diff
@@ -106,23 +106,27 @@ The `screen` module has the following methods:
...
 
 ### `screen.getPrimaryDisplay()`
 
-Returns the primary display.
+Returns `Display` - The primary display.
 
 ### `screen.getAllDisplays()`
 
-Returns an array of displays that are currently available.
+Returns `Display[]` - An array of displays that are currently available.
```
Comment (enlight) [inline docs/api/screen.md]: Returns an array of Display objects representing the displays that are currently available.

File: docs/api/session.md [hunk 5]
```diff
@@ -27,7 +27,7 @@ The `session` module has the following methods:
 * `options` Object
   * `cache` Boolean - Whether to enable cache.
 
-Returns a `Session` instance from `partition` string. When there is an existing
+Returns `Session` - A session instance from `partition` string. When there is an existing
```
Comment (enlight) [inline docs/api/session.md]: Returns a `Session` instance for the given `partition`.

File: docs/api/session.md [hunk 6]
```diff
@@ -46,7 +46,7 @@ The `session` module has the following properties:
 
 ### `session.defaultSession`
 
-Returns the default session object of the app.
+A `Session` object, the default session object of the app.
```
Comment (enlight) [inline docs/api/session.md]: This property is read-only right? If so, should document it as:

Returns the `Session` object for the default session of the app.

File: docs/api/web-contents.md [hunk 7]
```diff
@@ -29,19 +29,19 @@ console.log(webContents)
 
 ### `webContents.getAllWebContents()`
 
-Returns an array of all `WebContents` instances. This will contain web contents
+Returns `WebContents[]` - An array of all `WebContents` instances. This will contain web contents
```
Comment (enlight) [inline docs/api/web-contents.md]: Returns an array of WebContents instances which correspond to the contents of all windows, webviews, opened devtools, and devtools extension background pages.

File: docs/api/web-contents.md [hunk 8]
```diff
@@ -1160,26 +1160,26 @@ one through the `'paint'` event.
 
 #### `contents.id`
 
-The unique ID of this WebContents.
+A Integer representing the unique ID of this WebContents.
```
Comment (enlight) [inline docs/api/web-contents.md]: Returns an Integer representing the unique ID of this WebContents instance.

Should generate: `readonly id: number`.

File: docs/api/web-frame.md [hunk 9]
```diff
@@ -109,6 +109,43 @@ this limitation.
 
 ### `webFrame.getResourceUsage()`
 
+Returns `Object`:
```
Comment (enlight) [inline docs/api/web-frame.md]: Returns an Object describing usage information of Blink's internal memory caches:

File: docs/api/window-open.md [hunk 10]
```diff
@@ -86,4 +86,4 @@ The `BrowserWindowProxy` object has the following instance properties:
 
 #### `win.closed`
 
-Set to true after the child window gets closed.
+A Boolean that is set to true after the child window gets closed.
```
Comment (enlight) [inline docs/api/window-open.md]: Returns `true` after the child window gets closed.

Assumption: The TypeScript generator maps `true` / `false` -> Boolean.

File: docs/styleguide.md (hunk 2 above)
Comment (enlight) [inline docs/styleguide.md]: AND: Read-only properties must start with "Returns (a|an|the) TYPE ..."

File: docs/api/browser-window.md [hunk 11]
```diff
@@ -467,23 +467,23 @@ The `BrowserWindow` class has the following static methods:
 
 #### `BrowserWindow.getAllWindows()`
 
-Returns an array of all opened browser windows.
+Returns `BrowserWindow[]` - An array of all opened browser windows.
```
Comment (enlight) [inline docs/api/browser-window.md]: Returns an array of BrowserWindow instances that correspond to all the existing browser windows.

File: docs/styleguide.md (hunk 3 above)
Comment (enlight) [inline docs/styleguide.md]: "Returns (a|an|the) `[TYPE]` - description"

Comment (enlight) [inline docs/styleguide.md]: Also, I think it should be possible to describe array types as "array of TYPE" and "TYPE array" in the docs.

File: docs/api/browser-window.md (hunk 1 above)
Comment (enlight) [inline docs/api/browser-window.md]: I've mentioned this below already, but I don't think we should be introducing any new syntax, just write "array of TYPE" or "TYPE array"... I think the parser can be modified to deal with both variants right @zeke (or just the first one if it makes things easier)?

Comment (MarshallOfSound) [inline docs/api/browser-window.md]: I don't think this is new syntax, this is pretty standard around the place.  In fact, this is the UML standard for representing collections of a type.

Comment (MarshallOfSound) [inline docs/api/browser-window.md]: Also, the parser shouldn't be modified to handle abstract modifiers to types.  The type should be fully contained, the less magic we have to add to the parser the better

File: docs/api/browser-window.md [hunk 12]
```diff
@@ -540,15 +540,15 @@ win.loadURL('https://github.com')
 
 #### `win.webContents`
 
-The `WebContents` object this window owns. All web page related events and
+A `WebContents` object this window owns. All web page related events and
```
Comment (MarshallOfSound) [inline docs/api/browser-window.md]: @enlight To maintain scope of this PR things like readonly should be left to a new set of documentation / parser PR's

File: docs/api/browser-window.md (hunk 11 above)
Comment (MarshallOfSound) [inline docs/api/browser-window.md]: I think we need to go back to #7337 if you disagree with the new format for representing these things.  It was approved by @zeke in that issue so perhaps you should bring that up with him before I start making these changes.

I think the key thing here is the difference between human readable documentation and clear / accurate / formatted documentation.  A think this a compromise between the two (both readable and follows an exact format).

Comment (enlight) [inline docs/api/browser-window.md]: Good point, I will comment in #7337.

File: docs/api/browser-window.md (hunk 12 above)
Comment (enlight) [inline docs/api/browser-window.md]: Yeah, this can be sorted out later.

File: docs/styleguide.md [hunk 13]
```diff
@@ -102,9 +102,10 @@ Using `autoUpdater` as an example:
 * [Static Methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static) must be listed under a `### Static Methods` chapter.
 * [Instance Methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#Prototype_methods) must be listed under an `### Instance Methods` chapter.
 * All methods that have a return value must start their description with "Returns `[TYPE]` - Return description"
+  * If the method returns an `Object` with a given structure that structure can be specified with a color followed by a newline then an unordered list of properties in the same style as function parameters
```
Comment (zeke) [inline docs/styleguide.md]: s/color/colon/

**Comment by zcbenz** [comment]
Nice :+1:

**Pull request merged**